pkg/storage/stores/shipper/downloads/table.go

```diff
@@ -118,11 +121,18 @@ func (t *Table) init(ctx context.Context) (err error) {
 
 	level.Debug(util.Logger).Log("msg", fmt.Sprintf("list of files to download for period %s: %s", t.name, objects))
 
+	// download the dbs parallelly
+	err = t.doParallelDownload(ctx, objects)
+	if err != nil {
+		return err
+	}
+
 	folderPath, err := t.folderPathForTable(true)
```

Comment:
I think this could be moved before the `t.doParallelDownload` and you could pass the folder path into that function to avoid having to call it here and also inside that function, WDYT?